server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.repositories.generate_json_response')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.manager_factory.repo_sync_schedule_manager')
+    def test_delete_sync_schedule(self, mock_manager, mock_resp):
+        """
+        Delete a single schedule.
+        """
+
+        mock_request = mock.MagicMock()
+        sync_resource = RepoSyncScheduleResourceView()
+        response = sync_resource.delete(mock_request, 'mock_repo', 'mock_importer', 'mock_schedule')
+        sync_resource.manager.delete.assert_called_once_with(
+            'mock_repo', 'mock_importer', 'mock_schedule')
+        mock_resp.assert_called_once_with(None)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.repositories.generate_json_response')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.manager_factory.repo_sync_schedule_manager')
+    def test_delete_sync_schedule_invalid_schedule(self, mock_manager, mock_resp):
+        """
+        Attempt to delete a scheduled sync passing an invalid schedule id.
+        """
+
+        mock_request = mock.MagicMock()
+        selfmanager = mock_manager.return_value
+        selfmanager.delete.side_effect = pulp_exceptions.InvalidValue('InvalidValue')
+        sync_resource = RepoSyncScheduleResourceView()
+        try:
+            sync_resource.delete(mock_request, 'mock_repo', 'mock_importer', 'mock_schedule')
+        except pulp_exceptions.MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised if url param is invalid")
+
+        self.assertEqual(response.http_status_code, 404)
+        self.assertTrue(response.error_code is error_codes.PLP0009)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch('pulp.server.webservices.views.repositories.generate_json_response')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.manager_factory.repo_sync_schedule_manager')
+    def test_update_sync_schedule_no_schedule_param(self, mock_manager, mock_reverse, mock_resp):
+        """
+        Attempt to update a dschedueld sync while missing the required schedule parameter.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'some': 'data'})
+        sync_resource = RepoSyncScheduleResourceView()
+        response = sync_resource.put(mock_request, 'mock_repo', 'mock_importer', 'mock_schedule')
+        sync_resource.manager.update.assert_called_once_with(
```

Comment:
`mock_manager.update` is not actually called. The manager is set in the parent class when `RepoSyncScheduleResourceView` is initialized, making this a different object. 

I could use `mock_manager().update` but I wanted to be clear that it was the mock that belongs to the `sync_resource` instance.